data/src/main/java/me/androidbox/data/remote/authentication/AuthenticationRepositoryImp.kt

```diff
@@ -0,0 +1,66 @@
+package me.androidbox.data.remote.authentication
+
+import kotlinx.coroutines.CancellationException
+import me.androidbox.data.remote.model.request.LoginRequestDto
+import me.androidbox.data.remote.model.request.RegistrationRequestDto
+import me.androidbox.data.remote.network.authentication.AuthenticationService
+import me.androidbox.domain.authentication.NetworkResponseState
+import me.androidbox.domain.authentication.model.LoginModel
+import me.androidbox.domain.authentication.remote.AuthenticationRepository
+import retrofit2.HttpException
+import javax.inject.Inject
+
+class AuthenticationRepositoryImp @Inject constructor(
+    private val authenticationService: AuthenticationService,
+) : AuthenticationRepository {
+
+    override suspend fun registerUser(fullName: String, email: String, password: String): NetworkResponseState<Unit> {
+        val registrationRequestDto = RegistrationRequestDto(
+            fullName = fullName,
+            email = email,
+            password = password
+        )
+
+        return try {
+            authenticationService.register(registrationRequestDto)
+            NetworkResponseState.Success(Unit)
+        }
+        catch(httpException: HttpException) {
+             NetworkResponseState.Failure(httpException)
+        }
+        catch (exception: Exception) {
+            if(exception is CancellationException) {
+                throw Exception()
+            }
+            NetworkResponseState.Failure(exception)
+        }
```

Comment:
This can be simplified to:

```
catch (e: CancellationException) {
    throw e
} catch (exception: Exception) {
    NetworkResponseState.Failure(exception)
}
```

Reply:
Also, when it's a cancellation exception, don't throw a new Exception by doing **Exception()** . You should re throw the current exception by doing **throw e** (e being the exception variable name)